src/controllers/AuthController.js

```diff
@@ -224,6 +226,71 @@ class AuthController {
     return ({ email, password });
   }
 
+  /**
+   * Verifies if the token passed is linked to an active user and
+   *  performs pre-checks for staff authorization.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {string} req.get - The function to retrieve header values.
+   * @returns {Promise<Object>} The result of the staff pre-check.
+   * @throws {Error} If there is an error during the pre-check process.
+   */
+  static async staffPreCheck(req) {
+    // verify token passed is linked to an active user
+    // extract the token from the header X-Token
+    const token = req.get('X-Token');
+    if (!token) {
+      return ({ error: 'Missing X-Token in the Authorization header' });
+    }
+    const staffId = await this.getUserID(token);
+    if (staffId.error) {
+      return ({ error: 'Unauthorized', msg: staffId.error });
+    }
+    // check if server is up before verifying
+    if (!await dbClient.isAlive()) {
+      return ({ error: 'Internal Database Server Error' });
+    }
+    // validate if the token and object from the request are same
+    const staff = await Staff.findById({ _id: staffId });
+    if (!staff) {
+      return ({ error: 'Unauthorized', msg: 'Token is not linked to any staff account' });
+    }
+    return { staff, token };
+  }
+
+  /**
+   * Performs attribute checks on the staff request data.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {Object} req.body - The body of the request.
+   * @returns {Promise<Object>} The result of the attribute checks.
+   * @throws {Error} If there is an error during the attribute checks.
+   */
+  static async staffAttrCheck(req) {
+    const data = req.body;
+    if (!data) {
+      return ({ error: 'Missing data for the request' });
+    }
+    // ensure data is an object type
+    if (typeof data !== 'object') {
```

Comment:
The custom methods for the `staff` model already perform this check where applicable, so you could actually just omit it to keep your code shorter. Where application, data validation is always initiated first in all the custom methods.

_Same goes for line 288 below. . ._